Automattic/simplenote-electron
# Improve auto updater

This PR replaces the built in electron auto updater with`electron-updater`. One of the major benefits is added support for windows updates (nsis) and works perfectly in sync with the `electron-builder` binaries.

We also don't longer need to maintain a custom API to manage releases. Releases are now handled directly via Github releases that are automatically created whenever a new tag is pushed.

### How to do dev/ux-testing
1. Make sure your package.json has a version lower than `1.2.0`
1. In project root, create the file `dev-app-update.yml` with the following content:
   ```yml
   # dev-app-update.yml
   owner: adlk
   repo: simplenote-test
   provider: github
   ```
1. Run `make start` – assuming you have already built the web app once. Otherwise, run `make dev`
1. Watch the console output of the main process to see something like.
   ```console
   Checking for update
   Found version 1.2.0 (url: Simplenote-macOS-1.2.0.zip, Simplenote-macOS-1.2.0.dmg)
   Downloading update from Simplenote-macOS-1.2.0.zip, Simplenote-macOS-1.2.0.dmg
   ```

Please keep in mind that you can't actually update the app as you are running a local and unsigned binary of electron. So in any case, you'll most likely get an error when closing the update dialog. Even though we are in dev mode, all status events will be triggered.

### How to do "production" testing
1. Grab a built binary for Mac or Windows from `CircleCI > Workflows >  Automattic > simplenote-electron > test/auto-updater (latest build) > artifacts (job) > artifacts`
1. Install the app to e.g. `/Applications`
1. Launch the app and wait for the auto updater to prompt
1. Update and check that you are on Simplenote 1.2.0 (via menu: Simplenote > About Simplenote)





More infos about electron-updater can be found [here](https://www.electron.build/auto-update).
